```
Error running mint automation
+ mkdir -p 14104-1d4dd69/gopath/src/github.com/minio
+ git -C 14104-1d4dd69/gopath/src/github.com/minio clone --quiet https://github.com/minio/minio.git
+ git -C 14104-1d4dd69/gopath/src/github.com/minio/minio remote add minio https://github.com/minio/minio.git
+ git -C 14104-1d4dd69/gopath/src/github.com/minio/minio fetch --quiet minio pull/14104/head:pr14104
+ git -C 14104-1d4dd69/gopath/src/github.com/minio/minio checkout --quiet pr14104
+ GO111MODULE=on
+ GOPATH=/home/mint-auto/mint-auto/14104-1d4dd69/gopath
+ make -C 14104-1d4dd69/gopath/src/github.com/minio/minio --quiet
go: downloading github.com/dustin/go-humanize v1.0.0
go: downloading cloud.google.com/go/storage v1.10.0
go: downloading github.com/minio/minio-go/v7 v7.0.21
go: downloading github.com/minio/cli v1.22.0
go: downloading google.golang.org/api v0.58.0
go: downloading github.com/json-iterator/go v1.1.12
go: downloading github.com/Azure/azure-storage-blob-go v0.10.0
go: downloading github.com/Azure/azure-pipeline-go v0.2.2
go: downloading github.com/colinmarc/hdfs/v2 v2.2.0
go: downloading github.com/minio/sha256-simd v1.0.0
go: downloading github.com/jcmturner/gokrb5/v8 v8.4.2
go: downloading golang.org/x/sys v0.0.0-20220114195835-da31bd327af9
go: downloading github.com/rs/dnscache v0.0.0-20211102005908-e0241e321417
go: downloading github.com/secure-io/sio-go v0.3.1
go: downloading github.com/minio/sio v0.3.0
go: downloading github.com/fatih/color v1.13.0
go: downloading github.com/klauspost/compress v1.13.6
go: downloading github.com/minio/highwayhash v1.0.2
go: downloading github.com/golang-jwt/jwt/v4 v4.1.0
go: downloading github.com/Shopify/sarama v1.27.2
go: downloading golang.org/x/crypto v0.0.0-20220112180741-5e0467b6c7ce
go: downloading github.com/minio/kes v0.14.0
go: downloading github.com/buger/jsonparser v1.1.1
go: downloading github.com/xdg/scram v0.0.0-20180814205039-7eeb5667e42c
go: downloading github.com/klauspost/reedsolomon v1.9.15
go: downloading github.com/miekg/dns v1.1.43
go: downloading github.com/gorilla/mux v1.8.0
go: downloading github.com/bcicen/jstream v1.0.1
go: downloading github.com/cespare/xxhash/v2 v2.1.2
go: downloading github.com/inconshreveable/mousetrap v1.0.0
go: downloading github.com/klauspost/readahead v1.4.0
go: downloading github.com/go-openapi/loads v0.21.0
go: downloading github.com/dchest/siphash v1.2.1
go: downloading github.com/hashicorp/golang-lru v0.5.4
go: downloading github.com/djherbis/atime v1.0.0
go: downloading github.com/minio/selfupdate v0.4.0
go: downloading github.com/bits-and-blooms/bloom/v3 v3.0.1
go: downloading github.com/coreos/go-oidc v2.1.0+incompatible
go: downloading github.com/minio/console v0.14.1
go: downloading github.com/philhofer/fwd v1.1.2-0.20210722190033-5c56ac6d0bb9
go: downloading github.com/google/uuid v1.3.0
go: downloading github.com/cheggaaa/pb v1.0.29
go: downloading github.com/prometheus/client_golang v1.11.0
go: downloading github.com/pierrec/lz4 v2.6.0+incompatible
go: downloading github.com/klauspost/pgzip v1.2.5
go: downloading github.com/mitchellh/go-homedir v1.1.0
go: downloading github.com/prometheus/client_model v0.2.0
go: downloading github.com/rs/cors v1.7.0
go: downloading github.com/minio/zipindex v0.2.1
go: downloading github.com/cosnicolaou/pbzip2 v1.0.1
go: downloading github.com/shirou/gopsutil/v3 v3.21.12
go: downloading github.com/tinylib/msgp v1.1.7-0.20211026165309-e818a1881b0e
go: downloading github.com/prometheus/procfs v0.7.3
go: downloading github.com/yargevad/filepathx v1.0.0
go: downloading go.etcd.io/etcd/api/v3 v3.5.0
go: downloading go.etcd.io/etcd/client/v3 v3.5.0
go: downloading github.com/zeebo/xxh3 v1.0.0
go: downloading go.uber.org/atomic v1.9.0
go: downloading golang.org/x/oauth2 v0.0.0-20210819190943-2bc19b11175f
go: downloading github.com/valyala/bytebufferpool v1.0.0
go: downloading github.com/beevik/ntp v0.3.0
go: downloading golang.org/x/time v0.0.0-20210723032227-1f47c861a9ac
go: downloading go.uber.org/zap v1.19.1
go: downloading github.com/go-ldap/ldap/v3 v3.2.4
go: downloading github.com/montanaflynn/stats v0.6.6
go: downloading github.com/ncw/directio v1.0.5
go: downloading github.com/coredns/coredns v1.4.0
go: downloading github.com/minio/simdjson-go v0.2.1
go: downloading github.com/eclipse/paho.mqtt.golang v1.3.5
go: downloading github.com/elastic/go-elasticsearch/v7 v7.12.0
go: downloading github.com/go-sql-driver/mysql v1.5.0
go: downloading github.com/gomodule/redigo v1.8.8
go: downloading github.com/lib/pq v1.9.0
go: downloading github.com/nats-io/nats.go v1.11.1-0.20210623165838-4b75fc59ae30
go: downloading github.com/nats-io/stan.go v0.8.3
go: downloading github.com/nsqio/go-nsq v1.0.8
go: downloading github.com/pkg/errors v0.9.1
go: downloading github.com/streadway/amqp v1.0.0
go: downloading golang.org/x/net v0.0.0-20220121210141-e204ce36a2ba
go: downloading github.com/mattn/go-ieproxy v0.0.1
go: downloading github.com/bits-and-blooms/bitset v1.2.0
go: downloading github.com/minio/csvparser v1.0.0
go: downloading github.com/minio/parquet-go v1.1.0
go: downloading github.com/alecthomas/participle v0.2.1
go: downloading github.com/golang/protobuf v1.5.2
go: downloading github.com/golang-jwt/jwt v3.2.2+incompatible
go: downloading github.com/mattn/go-colorable v0.1.10
go: downloading github.com/mattn/go-isatty v0.0.14
go: downloading github.com/mattn/go-runewidth v0.0.13
go: downloading github.com/xdg/stringprep v1.0.0
go: downloading github.com/klauspost/cpuid/v2 v2.0.9
go: downloading github.com/gogo/protobuf v1.3.2
go: downloading go.etcd.io/etcd/client/pkg/v3 v3.5.0
go: downloading google.golang.org/grpc v1.41.0
go: downloading github.com/golang/snappy v0.0.3
go: downloading github.com/nats-io/nuid v1.0.1
go: downloading google.golang.org/genproto v0.0.0-20210928142010-c7af6a1a74c9
go: downloading github.com/coreos/go-semver v0.3.0
go: downloading github.com/rivo/uniseg v0.2.0
go: downloading golang.org/x/text v0.3.7
go: downloading golang.org/x/sync v0.0.0-20210220032951-036812b2e83c
go: downloading github.com/pquerna/cachecontrol v0.0.0-20171018203845-0dec1b30a021
go: downloading gopkg.in/square/go-jose.v2 v2.3.1
go: downloading github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd
go: downloading github.com/modern-go/reflect2 v1.0.2
go: downloading github.com/go-asn1-ber/asn1-ber v1.5.1
go: downloading github.com/Azure/go-ntlmssp v0.0.0-20200615164410-66371956d46c
go: downloading github.com/gorilla/websocket v1.4.2
go: downloading github.com/go-openapi/analysis v0.20.1
go: downloading github.com/go-openapi/spec v0.20.4
go: downloading github.com/go-openapi/swag v0.19.15
go: downloading google.golang.org/protobuf v1.27.1
go: downloading github.com/coreos/go-systemd/v22 v22.3.2
go: downloading github.com/prometheus/common v0.31.1
go: downloading github.com/beorn7/perks v1.0.1
go: downloading cloud.google.com/go v0.94.1
go: downloading go.uber.org/multierr v1.7.0
go: downloading github.com/hashicorp/go-uuid v1.0.2
go: downloading github.com/jcmturner/gofork v1.0.0
go: downloading github.com/jcmturner/dnsutils/v2 v2.0.0
go: downloading github.com/jcmturner/aescts/v2 v2.0.0
go: downloading github.com/apache/thrift v0.15.0
go: downloading github.com/rs/xid v1.3.0
go: downloading github.com/tidwall/sjson v1.2.3
go: downloading github.com/tidwall/gjson v1.11.0
go: downloading github.com/jcmturner/rpc/v2 v2.0.3
go: downloading gopkg.in/ini.v1 v1.66.3
go: downloading github.com/minio/md5-simd v1.1.2
go: downloading github.com/nats-io/nkeys v0.3.0
go: downloading gopkg.in/yaml.v2 v2.4.0
go: downloading github.com/mailru/easyjson v0.7.6
go: downloading gopkg.in/jcmturner/gokrb5.v7 v7.5.0
go: downloading github.com/eapache/queue v1.1.0
go: downloading github.com/eapache/go-resiliency v1.2.0
go: downloading github.com/davecgh/go-spew v1.1.1
go: downloading github.com/rcrowley/go-metrics v0.0.0-20200313005456-10cdbea86bc0
go: downloading github.com/eapache/go-xerial-snappy v0.0.0-20180814174437-776d5712da21
go: downloading github.com/jcmturner/goidentity/v6 v6.0.1
go: downloading github.com/tidwall/pretty v1.2.0
go: downloading github.com/tidwall/match v1.1.1
go: downloading github.com/josharian/intern v1.0.0
go: downloading github.com/matttproud/golang_protobuf_extensions v1.0.2-0.20181231171920-c182affec369
go: downloading github.com/go-openapi/strfmt v0.21.1
go: downloading github.com/go-openapi/jsonpointer v0.19.5
go: downloading github.com/go-openapi/jsonreference v0.19.6
go: downloading github.com/PuerkitoBio/purell v1.1.1
go: downloading github.com/PuerkitoBio/urlesc v0.0.0-20170810143723-de5bf2ad4578
go: downloading github.com/asaskevich/govalidator v0.0.0-20200907205600-7a23bdc65eef
go: downloading github.com/mitchellh/mapstructure v1.4.1
go: downloading github.com/go-openapi/errors v0.20.1
go: downloading go.mongodb.org/mongo-driver v1.7.5
go: downloading github.com/oklog/ulid v1.3.1
go: downloading github.com/googleapis/gax-go/v2 v2.1.1
go: downloading go.opencensus.io v0.23.0
go: downloading github.com/golang/groupcache v0.0.0-20200121045136-8c9f03a8e57e
go: downloading github.com/go-stack/stack v1.8.0
go: downloading github.com/go-openapi/runtime v0.21.0
go: downloading github.com/jessevdk/go-flags v1.4.0
go: downloading github.com/minio/mc v0.0.0-20211207230606-23a05f5a17f2
go: downloading github.com/unrolled/secure v1.0.9
go: downloading k8s.io/client-go v0.21.1
go: downloading github.com/minio/operator v0.0.0-20220110040724-a5d59a342b7f
go: downloading github.com/minio/direct-csi v1.3.5-0.20210601185811-f7776f7961bf
go: downloading github.com/go-openapi/validate v0.20.3
go: downloading github.com/docker/go-units v0.4.0
go: downloading k8s.io/apimachinery v0.21.1
go: downloading github.com/mb0/glob v0.0.0-20160210091149-1eb79d2de6c4
go: downloading k8s.io/kube-openapi v0.0.0-20210305001622-591a79e4bda7
go: downloading github.com/golang/glog v0.0.0-20160126235308-23def4e6c14b
go: downloading github.com/briandowns/spinner v1.16.0
go: downloading maze.io/x/duration v0.0.0-20160924141736-faac084b6075
go: downloading github.com/minio/filepath v1.0.0
go: downloading github.com/pkg/xattr v0.4.3
go: downloading github.com/minio/colorjson v1.0.1
go: downloading github.com/pkg/profile v1.6.0
go: downloading github.com/rjeczalik/notify v0.9.2
go: downloading github.com/posener/complete v1.2.3
go: downloading gopkg.in/h2non/filetype.v1 v1.0.5
go: downloading golang.org/x/term v0.0.0-20210927222741-03fcf44c2211
go: downloading github.com/tklauser/go-sysconf v0.3.9
go: downloading github.com/hashicorp/go-multierror v1.1.1
go: downloading github.com/hashicorp/errwrap v1.1.0
go: downloading github.com/tklauser/numcpus v0.3.0
go: downloading sigs.k8s.io/controller-runtime v0.8.0
go: downloading k8s.io/api v0.21.1
go: downloading k8s.io/klog/v2 v2.8.0
go: downloading github.com/go-logr/logr v0.4.0
go: downloading sigs.k8s.io/yaml v1.2.0
go: downloading github.com/google/gofuzz v1.1.0
go: downloading sigs.k8s.io/structured-merge-diff/v4 v4.1.0
go: downloading github.com/google/go-cmp v0.5.6
go: downloading gopkg.in/inf.v0 v0.9.1
go: downloading gopkg.in/jcmturner/dnsutils.v1 v1.0.1
go: downloading gopkg.in/jcmturner/aescts.v1 v1.0.1
go: downloading github.com/emicklei/go-restful v2.9.5+incompatible
go: downloading gopkg.in/jcmturner/rpc.v1 v1.1.0
go: downloading k8s.io/utils v0.0.0-20201110183641-67b214c5f920
go: downloading github.com/googleapis/gnostic v0.5.1
go: downloading gopkg.in/yaml.v3 v3.0.0-20210107192922-496545a6307b
../../../../pkg/mod/github.com/minio/operator@v0.0.0-20220110040724-a5d59a342b7f/pkg/apis/minio.min.io/v1/helper.go:49:2: github.com/minio/madmin-go@v1.2.8: replacement directory ../madmin-go does not exist
cmd/acl-handlers.go:28:2: github.com/minio/pkg@v1.1.15: replacement directory ../pkg does not exist
../../../../pkg/mod/github.com/minio/console@v0.14.1/restapi/user_session.go:28:2: github.com/minio/pkg@v1.1.15: replacement directory ../pkg does not exist
../../../../pkg/mod/github.com/minio/console@v0.14.1/pkg/certs/certs.go:35:2: github.com/minio/pkg@v1.1.15: replacement directory ../pkg does not exist
../../../../pkg/mod/github.com/minio/colorjson@v1.0.1/colors.go:9:2: github.com/minio/pkg@v1.1.15: replacement directory ../pkg does not exist
internal/config/cache/config.go:27:2: github.com/minio/pkg@v1.1.15: replacement directory ../pkg does not exist
../../../../pkg/mod/github.com/minio/console@v0.14.1/pkg/certs/certs.go:36:2: github.com/minio/pkg@v1.1.15: replacement directory ../pkg does not exist
../../../../pkg/mod/github.com/minio/mc@v0.0.0-20211207230606-23a05f5a17f2/cmd/admin-user-svcacct-add.go:32:2: github.com/minio/pkg@v1.1.15: replacement directory ../pkg does not exist
../../../../pkg/mod/github.com/minio/mc@v0.0.0-20211207230606-23a05f5a17f2/cmd/subnet-utils.go:39:2: github.com/minio/pkg@v1.1.15: replacement directory ../pkg does not exist
../../../../pkg/mod/github.com/minio/mc@v0.0.0-20211207230606-23a05f5a17f2/cmd/client-s3.go:53:2: github.com/minio/pkg@v1.1.15: replacement directory ../pkg does not exist
internal/logger/target/kafka/kafka.go:32:2: github.com/minio/pkg@v1.1.15: replacement directory ../pkg does not exist
../../../../pkg/mod/github.com/minio/mc@v0.0.0-20211207230606-23a05f5a17f2/cmd/config-fix.go:29:2: github.com/minio/pkg@v1.1.15: replacement directory ../pkg does not exist
cmd/peer-rest-server.go:43:2: github.com/minio/pkg@v1.1.15: replacement directory ../pkg does not exist
cmd/server-rlimit.go:25:2: github.com/minio/pkg@v1.1.15: replacement directory ../pkg does not exist
../../../../pkg/mod/github.com/minio/mc@v0.0.0-20211207230606-23a05f5a17f2/cmd/main.go:40:2: github.com/minio/pkg@v1.1.15: replacement directory ../pkg does not exist
../../../../pkg/mod/github.com/minio/mc@v0.0.0-20211207230606-23a05f5a17f2/cmd/find.go:37:2: github.com/minio/pkg@v1.1.15: replacement directory ../pkg does not exist
../../../../pkg/mod/github.com/minio/mc@v0.0.0-20211207230606-23a05f5a17f2/cmd/main.go:41:2: github.com/minio/pkg@v1.1.15: replacement directory ../pkg does not exist
make: *** [build] Error 1

```

--- a/cmd/admin-handlers-site-replication.go
+++ b/cmd/admin-handlers-site-replication.go
@@ -253,18 +253,6 @@ func (a adminAPIHandlers) SRPeerReplicateBucketItem(w http.ResponseWriter, r *ht
 	}
 }
 
-// SiteReplicationDisable - PUT /minio/admin/v3/site-replication/disable
-func (a adminAPIHandlers) SiteReplicationDisable(w http.ResponseWriter, r *http.Request) {
-	ctx := newContext(r, w, "SiteReplicationDisable")
-
-	defer logger.AuditLog(ctx, w, r, mustGetClaimsFromToken(r))
-
-	objectAPI, _ := validateAdminReq(ctx, w, r, iampolicy.SiteReplicationDisableAction)
-	if objectAPI == nil {
-		return
-	}
-}
-
 // SiteReplicationInfo - GET /minio/admin/v3/site-replication/info
 func (a adminAPIHandlers) SiteReplicationInfo(w http.ResponseWriter, r *http.Request) {
 	ctx := newContext(r, w, "SiteReplicationInfo")
@@ -313,7 +301,6 @@ func parseJSONBody(ctx context.Context, body io.Reader, v interface{}, encryptio
 			Code:  ErrSiteReplicationInvalidRequest,
 		}
 	}
-
 	if encryptionKey != "" {
 		data, err = madmin.DecryptData(encryptionKey, bytes.NewReader(data))
 		if err != nil {
@@ -324,7 +311,6 @@ func parseJSONBody(ctx context.Context, body io.Reader, v interface{}, encryptio
 			}
 		}
 	}
-
 	return json.Unmarshal(data, v)
 }
 
@@ -447,4 +433,60 @@ func getSRStatusOptions(r *http.Request) (opts madmin.SRStatusOptions) {
 	opts.Entity = madmin.GetSREntityType(q.Get("entity"))
 	opts.EntityValue = q.Get("entityvalue")
 	return
+}
+
+// SiteReplicationRemove - PUT /minio/admin/v3/site-replication/remove
+func (a adminAPIHandlers) SiteReplicationRemove(w http.ResponseWriter, r *http.Request) {
+	ctx := newContext(r, w, "SiteReplicationRemove")
+
+	defer logger.AuditLog(ctx, w, r, mustGetClaimsFromToken(r))
+
+	objectAPI, _ := validateAdminReq(ctx, w, r, iampolicy.SiteReplicationRemoveAction)
+	if objectAPI == nil {
+		return
+	}
+	var rreq madmin.SRRemoveReq
+	err := parseJSONBody(ctx, r.Body, &rreq, "")
+	if err != nil {
+		writeErrorResponseJSON(ctx, w, toAdminAPIErr(ctx, err), r.URL)
+		return
+	}
+	status, err := globalSiteReplicationSys.RemovePeerCluster(ctx, objectAPI, rreq)
+	if err != nil {
+		logger.LogIf(ctx, err)
+		writeErrorResponseJSON(ctx, w, toAdminAPIErr(ctx, err), r.URL)
+		return
+	}
+
+	body, err := json.Marshal(status)
+	if err != nil {
+		writeErrorResponseJSON(ctx, w, toAdminAPIErr(ctx, err), r.URL)
+		return
+	}
+	writeSuccessResponseJSON(w, body)
+}
+
+// SRPeerRemove - PUT /minio/admin/v3/site-replication/peer/remove
+//
+// used internally to tell current cluster to update endpoint for peer
+func (a adminAPIHandlers) SRPeerRemove(w http.ResponseWriter, r *http.Request) {
+	ctx := newContext(r, w, "SRPeerRemove")
+	defer logger.AuditLog(ctx, w, r, mustGetClaimsFromToken(r))
+
+	objectAPI, _ := validateAdminReq(ctx, w, r, iampolicy.SiteReplicationRemoveAction)
+	if objectAPI == nil {
+		return
+	}
+
+	var req madmin.SRRemoveReq
+	if err := parseJSONBody(ctx, r.Body, &req, ""); err != nil {
+		writeErrorResponseJSON(ctx, w, toAdminAPIErr(ctx, err), r.URL)
+		return
+	}
+
+	if err := globalSiteReplicationSys.InternalRemoveReq(ctx, objectAPI, req); err != nil {
+		logger.LogIf(ctx, err)
+		writeErrorResponseJSON(ctx, w, toAdminAPIErr(ctx, err), r.URL)
+		return
+	}
 }

--- a/cmd/admin-router.go
+++ b/cmd/admin-router.go
@@ -197,7 +197,7 @@ func registerAdminRouter(router *mux.Router, enableConfigOps bool) {
 
 			// Cluster Replication APIs
 			adminRouter.Methods(http.MethodPut).Path(adminVersion + "/site-replication/add").HandlerFunc(gz(httpTraceHdrs(adminAPI.SiteReplicationAdd)))
-			adminRouter.Methods(http.MethodPut).Path(adminVersion + "/site-replication/disable").HandlerFunc(gz(httpTraceHdrs(adminAPI.SiteReplicationDisable)))
+			adminRouter.Methods(http.MethodPut).Path(adminVersion + "/site-replication/remove").HandlerFunc(gz(httpTraceHdrs(adminAPI.SiteReplicationRemove)))
 			adminRouter.Methods(http.MethodGet).Path(adminVersion + "/site-replication/info").HandlerFunc(gz(httpTraceHdrs(adminAPI.SiteReplicationInfo)))
 			adminRouter.Methods(http.MethodGet).Path(adminVersion + "/site-replication/metainfo").HandlerFunc(gz(httpTraceHdrs(adminAPI.SiteReplicationMetaInfo)))
 			adminRouter.Methods(http.MethodGet).Path(adminVersion + "/site-replication/status").HandlerFunc(gz(httpTraceHdrs(adminAPI.SiteReplicationStatus)))
@@ -209,6 +209,7 @@ func registerAdminRouter(router *mux.Router, enableConfigOps bool) {
 			adminRouter.Methods(http.MethodGet).Path(adminVersion + "/site-replication/peer/idp-settings").HandlerFunc(gz(httpTraceHdrs(adminAPI.SRPeerGetIDPSettings)))
 			adminRouter.Methods(http.MethodPut).Path(adminVersion + "/site-replication/edit").HandlerFunc(gz(httpTraceHdrs(adminAPI.SiteReplicationEdit)))
 			adminRouter.Methods(http.MethodPut).Path(adminVersion + "/site-replication/peer/edit").HandlerFunc(gz(httpTraceHdrs(adminAPI.SRPeerEdit)))
+			adminRouter.Methods(http.MethodPut).Path(adminVersion + "/site-replication/peer/remove").HandlerFunc(gz(httpTraceHdrs(adminAPI.SRPeerRemove)))
 		}
 
 		if globalIsDistErasure {

--- a/cmd/site-replication.go
+++ b/cmd/site-replication.go
@@ -1913,8 +1913,7 @@ func (c *SiteReplicationSys) RemovePeerCluster(ctx context.Context, objectAPI Ob
 			}, errSRPeerResp(fmt.Errorf("unable to update peer %s: %w", c.state.Peers[dID].Name, err))
 		}
 	}
-	// Other than handling existing buckets, we can now save the cluster
-	// replication configuration state.
+	// Update cluster state
 	var state srState
 	if len(updatedPeers) > 1 {
 		state = srState{
@@ -1923,7 +1922,6 @@ func (c *SiteReplicationSys) RemovePeerCluster(ctx context.Context, objectAPI Ob
 			ServiceAccountAccessKey: info.ServiceAccountAccessKey,
 		}
 	}
-	// need to delete replication rule
 	if err = c.saveToDisk(ctx, state); err != nil {
 		return madmin.ReplicateRemoveStatus{
 			Status:    madmin.ReplicateRemoveStatusPartial,

--- a/cmd/site-replication.go
+++ b/cmd/site-replication.go
@@ -255,7 +255,7 @@ func (c *SiteReplicationSys) saveToDisk(ctx context.Context, state srState) erro
 	c.Lock()
 	defer c.Unlock()
 	c.state = state
-	c.enabled = true
+	c.enabled = len(c.state.Peers) != 0
 	return nil
 }
 
@@ -1846,6 +1846,198 @@ func (c *SiteReplicationSys) isEnabled() bool {
 	c.RLock()
 	defer c.RUnlock()
 	return c.enabled
+}
+
+// RemovePeerCluster - removes one or more clusters from site replication configuration.
+func (c *SiteReplicationSys) RemovePeerCluster(ctx context.Context, objectAPI ObjectLayer, rreq madmin.SRRemoveReq) (st madmin.ReplicateRemoveStatus, err error) {
+	if !c.isEnabled() {
+		return st, errSRNotEnabled
+	}
+	info, err := c.GetClusterInfo(ctx)
+	if err != nil {
+		return st, errSRBackendIssue(err)
+	}
+	peerMap := make(map[string]madmin.PeerInfo)
+	var rmvEndpoints []string
+	siteNames := rreq.SiteNames
+	updatedPeers := make(map[string]madmin.PeerInfo)
+
+	for _, pi := range info.Sites {
+		updatedPeers[pi.DeploymentID] = pi
+		peerMap[pi.Name] = pi
+		if rreq.RemoveAll {
+			siteNames = append(siteNames, pi.Name)
+		}
+	}
+	for _, s := range siteNames {
+		info, ok := peerMap[s]
+		if !ok {
+			return st, errSRInvalidRequest(fmt.Errorf("Site %s not found in site replication configuration", s))
+		}
+		rmvEndpoints = append(rmvEndpoints, info.Endpoint)
+		delete(updatedPeers, info.DeploymentID)
+	}
+	var wg sync.WaitGroup
+	errs := make(map[string]error, len(c.state.Peers))
+
+	for _, v := range info.Sites {
+		wg.Add(1)
+		if v.DeploymentID == globalDeploymentID {
+			go func() {
+				defer wg.Done()
+				err := c.RemoveRemoteTargetsForEndpoint(ctx, objectAPI, rmvEndpoints, false)
+				errs[globalDeploymentID] = err
+			}()
+			continue
+		}
+		go func(pi madmin.PeerInfo) {
+			defer wg.Done()
+			admClient, err := c.getAdminClient(ctx, pi.DeploymentID)
+			if err != nil {
+				errs[pi.DeploymentID] = errSRPeerResp(fmt.Errorf("unable to create admin client for %s: %w", pi.Name, err))
+				return
+			}
+			if _, err = admClient.SRPeerRemove(ctx, rreq); err != nil {
+				errs[pi.DeploymentID] = errSRPeerResp(fmt.Errorf("unable to update peer %s: %w", pi.Name, err))
+				return
+			}
+		}(v)
+	}
+	wg.Wait()
+
+	for dID, err := range errs {
+		if err != nil {
+			return madmin.ReplicateRemoveStatus{
+				ErrDetail: err.Error(),
+				Status:    madmin.ReplicateRemoveStatusPartial,
+			}, errSRPeerResp(fmt.Errorf("unable to update peer %s: %w", c.state.Peers[dID].Name, err))
+		}
+	}
+	// Other than handling existing buckets, we can now save the cluster
+	// replication configuration state.
+	var state srState
+	if len(updatedPeers) > 1 {
+		state = srState{
+			Name:                    info.Name,
+			Peers:                   updatedPeers,
+			ServiceAccountAccessKey: info.ServiceAccountAccessKey,
+		}
+	}
+	// need to delete replication rule
+	if err = c.saveToDisk(ctx, state); err != nil {
+		return madmin.ReplicateRemoveStatus{
+			Status:    madmin.ReplicateRemoveStatusPartial,
+			ErrDetail: fmt.Sprintf("unable to save cluster-replication state on local: %v", err),
+		}, nil
+	}
+
+	return madmin.ReplicateRemoveStatus{
+		Status: madmin.ReplicateRemoveStatusSuccess,
+	}, nil
+}
+
+// InternalRemoveReq - sends an unlink request to peer cluster to remove one or more sites
+// from the site replication configuration.
+func (c *SiteReplicationSys) InternalRemoveReq(ctx context.Context, objectAPI ObjectLayer, rreq madmin.SRRemoveReq) error {
+	ourName := ""
+	peerMap := make(map[string]madmin.PeerInfo)
+	updatedPeers := make(map[string]madmin.PeerInfo)
+	siteNames := rreq.SiteNames
+
+	for _, p := range c.state.Peers {
+		peerMap[p.Name] = p
+		if p.DeploymentID == globalDeploymentID {
+			ourName = p.Name
+		}
+		updatedPeers[p.DeploymentID] = p
+		if rreq.RemoveAll {
+			siteNames = append(siteNames, p.Name)
+		}
+	}
+	var rmvEndpoints []string
+	var unlinkSelf bool
+
+	for _, s := range siteNames {
+		info, ok := peerMap[s]
+		if !ok {
+			return fmt.Errorf("Site %s not found in site replication configuration", s)
+		}
+		if info.DeploymentID == globalDeploymentID {
+			unlinkSelf = true
+			continue
+		}
+		delete(updatedPeers, info.DeploymentID)
+		rmvEndpoints = append(rmvEndpoints, info.Endpoint)
+	}
+	if err := c.RemoveRemoteTargetsForEndpoint(ctx, objectAPI, rmvEndpoints, unlinkSelf); err != nil {
+		return err
+	}
+	var state srState
+	if !unlinkSelf {
+		state = srState{
+			Name:                    c.state.Name,
+			Peers:                   updatedPeers,
+			ServiceAccountAccessKey: c.state.ServiceAccountAccessKey,
+		}
+	}
+
+	if err := c.saveToDisk(ctx, state); err != nil {
+		return errSRBackendIssue(fmt.Errorf("unable to save cluster-replication state to disk on %s: %v", ourName, err))
+	}
+	return nil
+}
+
+// RemoveRemoteTargetsForEndpoint removes replication targets corresponding to endpoint
+func (c *SiteReplicationSys) RemoveRemoteTargetsForEndpoint(ctx context.Context, objectAPI ObjectLayer, endpoints []string, unlinkSelf bool) (err error) {
+	targets := globalBucketTargetSys.ListTargets(ctx, "", string(madmin.ReplicationService))
+	m := make(map[string]madmin.BucketTarget)
+	for _, t := range targets {
+		for _, endpoint := range endpoints {
+			ep, _ := url.Parse(endpoint)
+			if t.Endpoint == ep.Host &&
+				t.Secure == (ep.Scheme == "https") &&
+				t.Type == madmin.ReplicationService {
+				m[t.Arn] = t
+			}
+		}
+		// all remote targets from self are to be delinked
+		if unlinkSelf {
+			m[t.Arn] = t
+		}
+	}
+	buckets, err := objectAPI.ListBuckets(ctx)
+	for _, b := range buckets {
+		config, err := globalBucketMetadataSys.GetReplicationConfig(ctx, b.Name)
+		if err != nil {
+			return err
+		}
+		var nRules []sreplication.Rule
+		for _, r := range config.Rules {
+			if _, ok := m[r.Destination.Bucket]; !ok {
+				nRules = append(nRules, r)
+			}
+		}
+		if len(nRules) > 0 {
+			config.Rules = nRules
+			configData, err := xml.Marshal(config)
+			if err != nil {
+				return err
+			}
+			if err = globalBucketMetadataSys.Update(ctx, b.Name, bucketReplicationConfig, configData); err != nil {
+				return err
+			}
+		} else {
+			if err := globalBucketMetadataSys.Update(ctx, b.Name, bucketReplicationConfig, nil); err != nil {
+				return err
+			}
+		}
+	}
+	for arn, t := range m {
+		if err := globalBucketTargetSys.RemoveTarget(ctx, t.SourceBucket, arn); err != nil {
+			return err
+		}
+	}
+	return
 }
 
 // Other helpers

